app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,8 @@
+class RecipesController < ApplicationController
+
+    def show
+        @recipe = Recipe.find(params[:id])
+        @total_cost = @recipe.ingredients.sum(:cost)
```

Comment:
Doing`sum` in the controller would be considered an MVC infraction. You want to abstract this method to the model and then test it.